homeassistant/components/device_tracker/icloud.py

```diff
@@ -1,100 +1,449 @@
+ATTR_GMTT_DURATION = 'gmtt_duration'
+ATTR_GMTT_ORIGIN = 'gmtt_origin'
+
+ICLOUDTRACKERS = {}
+
+_CONFIGURING = {}
+
+DEVICESTATUSSET = ['features', 'maxMsgChar', 'darkWake', 'fmlyShare',
+                   'deviceStatus', 'remoteLock', 'activationLocked',
+                   'deviceClass', 'id', 'deviceModel', 'rawDeviceModel',
+                   'passcodeLength', 'canWipeAfterLock', 'trackingInfo',
+                   'location', 'msg', 'batteryLevel', 'remoteWipe',
+                   'thisDevice', 'snd', 'prsId', 'wipeInProgress',
+                   'lowPowerMode', 'lostModeEnabled', 'isLocating',
+                   'lostModeCapable', 'mesg', 'name', 'batteryStatus',
+                   'lockedTimestamp', 'lostTimestamp', 'locationCapable',
+                   'deviceDisplayName', 'lostDevice', 'deviceColor',
+                   'wipedTimestamp', 'modelDisplayName', 'locationEnabled',
+                   'isMac', 'locFoundEnabled']
+
+DEVICESTATUSCODES = {'200': 'online', '201': 'offline', '203': 'pending',
+                     '204': 'unregistered'}
+
+SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ACCOUNTNAME): vol.All(cv.ensure_list, [cv.slugify]),
+    vol.Optional(ATTR_DEVICENAME): cv.slugify,
+    vol.Optional(ATTR_INTERVAL): cv.positive_int,
+})
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_USERNAME): vol.Coerce(str),
-    vol.Required(CONF_PASSWORD): vol.Coerce(str),
-    vol.Optional(CONF_INTERVAL, default=8): vol.All(vol.Coerce(int),
-                                                    vol.Range(min=1))
-    })
-
-
-def setup_scanner(hass, config, see):
-    """Setup the iCloud Scanner."""
-    from pyicloud import PyiCloudService
-    from pyicloud.exceptions import PyiCloudFailedLoginException
-    from pyicloud.exceptions import PyiCloudNoDevicesException
-    logging.getLogger("pyicloud.base").setLevel(logging.WARNING)
-
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-
-    try:
-        _LOGGER.info('Logging into iCloud Account')
-        # Attempt the login to iCloud
-        api = PyiCloudService(username, password, verify=True)
-    except PyiCloudFailedLoginException as error:
-        _LOGGER.exception('Error logging into iCloud Service: %s', error)
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(ATTR_ACCOUNTNAME): cv.slugify,
+    vol.Optional(CONF_GMTT, default={}):
+        vol.Schema({cv.slugify: cv.entity_id}),
+})
+
+
+def setup_scanner(hass, config: dict, see):
+    """Set up the iCloud Scanner."""
+    # pylint: disable=too-many-locals
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    account = config.get(CONF_ACCOUNTNAME, slugify(username.partition('@')[0]))
+
+    googletraveltime = config.get(CONF_GMTT)
+
+    icloudaccount = Icloud(hass, username, password, account, googletraveltime,
+                           see)
+    if icloudaccount:
```

Comment:
This will always be the case? Can you instead try to add some validation of credentials ?